consensus/src/experimental/buffer_manager.rs

```diff
@@ -524,9 +526,45 @@ impl StateManager {
         Ok(())
     }
 
+    /// this function retries all the items until the signing root
+    /// note that there might be other signed items after the signing root
+    async fn retry_broadcasting_commit_votes(&mut self) -> anyhow::Result<()> {
+        let mut cursor = self.buffer.head.clone();
+        while cursor.is_some() && !link_eq(&cursor, &self.signing_root) {
+            // we move forward before sending the message
+            // just in case the buffer becomes empty during await.
+            let next_cursor = get_next(&cursor);
+            {
+                let buffer_item = get_elem(&cursor);
+                match &*buffer_item {
+                    BufferItem::Aggregated(aggregated) => {
```

Comment:
why we retry on already aggregated items?

Reply:
to help other nodes

Reply:
that's not how we do it, we broadcast sync info to help other nodes

Reply:
But that will trigger the sync process (expensive?). It is possible that many nodes just needs one more signature to proceed. So I think broadcasting the in-buffer aggregated items can improve the situation. This is demonstrated in the previous cluster experiments that letting the commit vote carry more signatures significantly stabilizes the throughput.

Reply:
sync process is not supposed to be expensive or complicated, (that's one of the thing this refactor is supposed to fix). the node should just forward the aggregated LI as part of the sync request and the node could properly pop the buffer.

Reply:
A sync process will get least involve 3 network messages (sync req, fetch req, fetch resp, and potentially chunk req, chunk resp), possible txn replay, and extra disk I/O; but blindly broadcasting aggregated items will waste network resource. Maybe we can deterministically select a single node to broadcast the aggregated items? 

I'll change it to signed items for now. Later we can collect some stats.